build-data-capturing-maven-samples/capture-thermal-throttling/maven-thermal-throttling.groovy

```diff
@@ -0,0 +1,134 @@
+import java.nio.charset.Charset
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+
+/**
+ * This Groovy script captures the thermal throttling level and adds it as a tag.
+ * Some parameters can be tweaked according to your build:
+ * - SAMPLING_INTERVAL_IN_SECONDS, frequency at which the capture command is run
+ * - THROTLLING_LEVEL, list of throttling levels and value ranges (to be compared with captured throttling average value)
+ *
+ * WARNING:
+ * - This is supported on MacOS only.
+ */
+
+def thermalThrottlingService = new ThermalThrottlingService()
+
+// start service
+thermalThrottlingService.start()
+
+buildScan.buildFinished(buildResult -> {
+  try {
+    // process results on build completion
+    thermalThrottlingService.processResults(buildScan)
+  } finally {
+    // stop service
+    thermalThrottlingService.stop()
+  }
+})
+
+// Thermal Throttling service implementation
+class ThermalThrottlingService {
+
+  /**
+   * Sampling interval can be adjusted according to total build time.
+   */
+  def static final SAMPLING_INTERVAL_IN_SECONDS = 5
+
+  /**
+   * Throttling levels by throttling average value.
+   */
+  def static final THROTTLING_LEVEL =
+          [
+                  [level: "THROTTLING_HIGH", range: 0..40],
+                  [level: "THROTTLING_MEDIUM", range: 40..80],
+                  [level: "THROTTLING_LOW", range: 80..100]
+          ]
+
+  def static final COMMAND_ARGS = ["pmset", "-g", "therm"]
+  def static final COMMAND_OUTPUT_PARSING_PATTERN = /CPU_Speed_Limit\s+=\s+/
+
+  def final scheduler
+  def final samples
+
+  ThermalThrottlingService() {
+    scheduler = Executors.newScheduledThreadPool(1)
+    samples = new ConcurrentLinkedQueue<Integer>()
+  }
+
+  void start() {
+    def osName = System.getProperty("os.name")
+    if(osName.contains("OS X") || osName.startsWith("Darwin")) {
```

Comment:
```suggestion
    if (osName.contains("OS X") || osName.startsWith("Darwin")) {
```

Reply:
also same comment here, can we quickly abort the script if we aren't on osX as opposed to checking in here?